docs/reference/query-dsl/nested-query.asciidoc

```diff
@@ -1,34 +1,99 @@
 [[query-dsl-nested-query]]
 === Nested Query
 
-Nested query allows to query nested objects / docs (see
-<<nested,nested mapping>>). The
-query is executed against the nested objects / docs as if they were
-indexed as separate docs (they are, internally) and resulting in the
-root parent doc (or parent nested mapping). Here is a sample mapping we
-will work with:
+A `nested` query allows you to query objects of the <<nested,nested datatype>>
+type, executing against each field in an array. You can query each field
+independently while maintaining the relationship between fields in a record,
+for example, a user's first name, last name, and address.
+
+As an example, let's use the same mapping from <<nested,nested datatype>>.
+This `nested` query will correctly select only the arrays that contain both
+the first name Alice and the last name White. Because the objects are mapped
+as nested, if you queried for user.first Alice and user.last Smith, this query
+would return no results.
```

Comment:
My preference is to reserve "we" for cases where Elastic is recommending or doing something. In other cases, let's (hah!) keep the focus on the reader (you) performing the actions. (For procedures, I prefer to use the imperative, but it doesn't make sense to present every example as a step-by-step procedure.) Also, I'm a big fan of just saying what things do, not what they will do or what would have happened. There are cases where timing is important, but in general sticking to present tense is preferable. 

I'm increasingly adopting a "less is more" attitude about cross references & would definitely avoid linking to the same topic more than once. I think I'd drop the second reference to nested datatype entirely--readers aren't likely to know/remember the mappings from that example--I'd just briefly describe the mappings to set the stage for the example. 

I'd probably dodge the "correctly selects" terminology, which kind of makes it sounds like there's a case where Elasticsearch returns incorrect results. Maybe flip it around a bit, "In the following example, using a nested query matches only those fans with the first name Alice and the last name White in the nested user field. If you search for `user.first` _Alice_ and `user.last` _Smith_, you won't get any hits."